web-common/src/features/sources/modal/AddDataForm.svelte

```diff
@@ -104,6 +129,16 @@
   // Emit the submitting state to the parent
   $: dispatch("submitting", { submitting });
 
+  // Generate YAML preview from form state
+  $: yamlPreview = (() => {
+    const values = useDsn ? $dsnForm : $paramsForm;
+    return compileConnectorYAML(connector, values);
```

Comment:
We'll have to handle both `formType === "source"` and `formType === "connector"`, else the preview will not match the generated file. (Unfortunately, this is an example of the complexity we want to unravel by moving to a consistent connector+model paradigm.) Check out the `submitAddSourceForm()` function to see how the `source` files are generated.